seaworthy/tests-core/test_logs.py

```diff
@@ -320,46 +317,30 @@ def run(self):
         # timeout.
         self.finished.wait(self.con._close_timeout)
         # Time to clean up.
-        self.sock.shutdown(socket.SHUT_RDWR)
-        self.sock.close()
-        self.client_sock.close()
-        self.con._feeders.discard(self)
+        self.q.put(None)
+        # self.con._feeders.discard(self)
 
 
 class FakeCancellableStream:
     """
     Fake CancellableStream that iterates over the log data.
     """
 
-    def __init__(self, feeder, client):
+    def __init__(self, feeder, q):
         self._feeder = feeder
-        self.raw = socket.SocketIO(client, 'rb')
+        self._q = q
 
     def __iter__(self):
         return self
 
     def __next__(self):
-        try:
-            return self._next()
-        except socket.error:
+        # Once the stream is finished, we set the queue to None to avoid
+        # accidentally blocking forever trying to read from it again.
```

Comment:
Does that mean we should have an `assert q != None` here?

Reply:
The `self._q.get()` serves the same purpose (and the comment is carefully positioned to show up in the pytest output), but an assertion instead of the comment is a much better plan. :-)